pkg/storage/v2/mysql/client.go

```diff
@@ -199,3 +207,29 @@ func (c *client) RunInTransaction(ctx context.Context, tx Transaction, f func()
 	}
 	return err
 }
+
+func (c *client) RunInTransactionV2(ctx *context.Context, f func(tx Transaction) error) error {
```

Comment:
Nice work! we don't have to call commit or create `Transaction` struct manually.

In my opinion, I see `tx` will be re-declared at the beginning, should this input become redundant and we don't create it manually in API or domain layer. WDYT?

Reply:
@hvn2k1 
Thank you for your comment.
Excuse me, but could you please provide a simple example of code that illustrates the content of your comments?

Reply:
```suggestion
func (c *client) RunInTransactionV2(ctx *context.Context, f func() error) error {
```

Oh I mean it would be like this, since tx will be re-declared in line 212

Reply:
That `tx` is defined as an argument of the callback function where transaction processing is implemented.

We are passing it to the storage that has not yet been refactored because it is necessary, but if it is no longer needed, we plan to change it so that it will not be passed to the callback function as you pointed out.
(Its contents are described in the [comments.](https://github.com/bucketeer-io/bucketeer/blob/update-storage-client-auto-ops/pkg/storage/v2/mysql/client.go#L111))

Examples of places that are still needed:
https://github.com/bucketeer-io/bucketeer/blob/update-storage-client-auto-ops/pkg/feature/api/feature.go#L1401

What do you think?

Reply:
Thank you for clear explanation!